eox_nelp/user_profile/api/v1/views.py

```diff
@@ -0,0 +1,79 @@
+"""Generic views for NELP user's profile application.
+
+Function-views:
+  - generate_otp: generate and send via SMS the OTP related a user. Saved in cache.
+  - validate_otp: Compare and check if the proposed OTP match the User OTP saved in cache.
+    If match, updates the profile phone_number.
+
+"""
+import logging
+
+from django.core.cache import cache
+from django.db import transaction
+from django.http import HttpResponseForbidden, JsonResponse
+from edx_rest_framework_extensions.auth.session.authentication import SessionAuthenticationAllowInactiveUser
+from rest_framework import status
+from rest_framework.decorators import api_view, authentication_classes, permission_classes
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+
+from eox_nelp.edxapp_wrapper.user_api import accounts, errors
+
+logger = logging.getLogger(__name__)
+
+
+@api_view(["POST"])
+@authentication_classes((
+    SessionAuthenticationAllowInactiveUser,
+))
+@permission_classes((IsAuthenticated,))
+def update_user_data(request):
+    """ View to update user's fields.
+    ## Usage
+
+    ### **POST** /eox-nelp/api/user-profile/v1/update-user-data/
```

Comment:
This is an update view and works with a Post request? Does it have to be that way?

Reply:
I think you implemented that, as mentioned before this is a refactor and I keep same functionality, you can check the first commit where github just marked the renamed file https://github.com/eduNEXT/eox-nelp/pull/168/commits/3e2a95981c552ab39405c25fe4be48bdbc86733f#diff-a79e9a949c512eb3de51c9ca381be97e75441032280b46f251fdd4118f8d6f1a

Reply:
Yes but I didnt use an endpoint with `update` wording. But for the moment,I will not delay with the forms

Reply:
But I had to use the update word because this uses a POST instead a PUT hahaha to be honest I tried to implement the PUT however the OTP mixin doesn't work with PUT, so changing that requires more things that a simple refactor

Reply:
Now it is very very clear. 
  I prefer how you treated it and avoid  giving more complexity to the code with a PUT